### PR TITLE
Me/dpc 4143 filter secrets from logs

### PR DESCRIPTION
**Pull request by MEspositoE14s**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4143

## 🛠 Changes

Added a secrets filter to DropWizard logs in dpc-api, dpc-attribution, dpc-aggregation and dpc-consent.
Secrets are configured in each module's application.yml file.

## ℹ️ Context

We had an issue where a secret was accidentally leaked, this should prevent that going forward.

## 🧪 Validation

- Ran locally and made sure secrets were filtered from the logs.  
- Added unit tests to verify the filter works as expected.

### CODE REVIEW

File: dpc-common/src/main/java/gov/cms/dpc/common/logging/filters/SecretLoggingFilter.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+				i.remove();
+			}
+		}
+
+		return new Filter<>() {
+			@Override
+			public FilterReply decide(ILoggingEvent event) {
+				for (String secret : secrets) {
+					if (event.getFormattedMessage().contains(envVars.get(secret))) {
+						logger.warn("Suppressing log, attempted to write " + secret + " in " + event.getLoggerName());
```
Comment (jdettmannnava) [inline dpc-common/src/main/java/gov/cms/dpc/common/logging/filters/SecretLoggingFilter.java:48]: This took me a second -- it made me think you were about to right QQT^IQ1qjkaasf; in the log!

Comment (MEspositoE14s) [reply dpc-common/src/main/java/gov/cms/dpc/common/logging/filters/SecretLoggingFilter.java:48]: I double checked this three different times to make sure I didn't accidentally do that!  

For the sake of slightly better clarity and not confusing the next person, renamed `secret` -> `secretName`.

File: dpc-common/pom.xml [hunk 2]
```diff
@@ -147,6 +147,10 @@
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-params</artifactId>
         </dependency>
+        <dependency>
+            <groupId>uk.org.webcompere</groupId>
```
Comment (jdettmannnava) [inline dpc-common/pom.xml:151]: How did you choose this?

Comment (MEspositoE14s) [reply dpc-common/pom.xml:151]: I was looking for a library that would make it easier to temporarily set env variables for unit tests and it did the job.  It's only brought in for tests, from the main .pom...
```
<dependency>
  <groupId>uk.org.webcompere</groupId>
  <artifactId>system-stubs-jupiter</artifactId>
  <version>2.1.6</version>
  <scope>test</scope>
</dependency>
```

Comment (MEspositoE14s) [reply dpc-common/pom.xml:151]: It seemed cleaner than mocking the system call to `System.getenv()` inside filter class.

**Review by jdettmannnava — APPROVED** [review]
LGTM

**Pull request merged**